### PR TITLE
Refactor the Dockerfile

### PR DESCRIPTION
**Pull request by christophermaier**
Streamline the Dockerfile, getting rid of things we never used, removing some unnecessary and costly operations, and leverage the Docker cache for quicker builds.

It's also smaller!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
-    apk del gcc g++ && \
-    rm -f /var/cache/apk/*
+# Really, we only need the cog directory to be owned by operable,
+# because (by default) that's where we write log files. None of the
+# actual scripts or library files need to be owned by operable.
+RUN chown -R operable /home/operable/cog
+
+COPY mix.exs mix.lock /home/operable/cog/
+COPY config/ /home/operable/cog/config/
+RUN mix deps.get
```
Comment (vanstee) [inline Dockerfile]: You may want to add `--only prod` here. We can also skip the archive check if you want, but not as big of a deal.

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+        libstdc++ && \
+    mix deps.compile && \
+    apk del .build_deps
+
+COPY emqttd_plugins/ /home/operable/cog/emqttd_plugins/
+COPY priv/ /home/operable/cog/priv/
+COPY web/ /home/operable/cog/web/
+COPY lib/ /home/operable/cog/lib/
+
+RUN mix compile
```
Comment (vanstee) [inline Dockerfile]: To speed things up you could add `--no-deps-check` and `--no-archive-check` here.

File: Makefile [hunk 3]
```diff
@@ -13,7 +13,6 @@ reset-db:
 docker:
 	scripts/get_cogctl.sh
 	docker build \
-		   --build-arg MIX_ENV=prod \
 		   -t $(DOCKER_IMAGE) .
```
Comment (vanstee) [inline Makefile]: Totally just a minor style thing, but this might be easier to read on a single line.

File: Dockerfile (hunk 1 above)
Comment (christophermaier) [inline Dockerfile]: 👍

File: Dockerfile (hunk 2 above)
Comment (christophermaier) [inline Dockerfile]: 👍

File: Makefile (hunk 3 above)
Comment (christophermaier) [inline Makefile]: 👍

**Pull request merged**